src/opustrainer/modifiers/placeholders.py

```diff
@@ -287,54 +283,100 @@ def __call__(self, line:str) -> str:
         """
 
         src, trg, alignment = line.strip().split('\t')
```

Comment:
Ah, the `strip()` here messes us up. I was assuming that if we wouldn't have any alignment info, we'd at least still have the `\t` at the end indicating an empty cell in the row. But `strip()` gets rid of that. I'll add a `warn_once` and then continue as if `alginments == []`.

Reply:
I think not having alignment info is a symptom of really bad mapping, so it's better to skip this example as to not confuse the model. This is why i'd prefer to skip this sentence completely.

Reply:
Ah, Hmm, that's a bit more tricky. Right now modifiers can't add extra sentences, or remove sentences. Okay if we move that to another issue to add support for that? We can make something that also works nicely with #21 .

Reply:
So, for now, i just parse `fail\t\fail` which at least can be read as a valid copy noise, but yeah, more in line with the noise modifier thingie.

Reply:
I think it will be fine the (empty) column is missing if guided-alignment is turned on, but will complain about it if it is turned off. Bit the wrong way around if you'd ask me.

The reading uses `util::splitTsv`:
https://github.com/marian-nmt/marian-dev/blob/68cc88f70c1c5011288cb0f9360dd5a8a68a0f7e/src/data/corpus.cpp#L103-L117

and `util::splitTsv` complains only if there are more fields than it exects:
https://github.com/marian-nmt/marian-dev/blob/68cc88f70c1c5011288cb0f9360dd5a8a68a0f7e/src/common/utils.cpp#L71-L88

Reply:
Lol our `DatasetReader` class skips any lines that have empty cells. So this line of code will only throw if a dataset contains a line that is missing the entire cell (i.e. doesn't have the expected number of `\t` characters.)

I'd still prefer to solve this through #29, i.e. change the modifiers to return a list of outputs and possibly an empty list if it didn't like the input.